### ISSUE TITLE
Incorrect 'defined but never used' warning for private class fields (#)

### ISSUE THREAD

**Issue by brupelo** [oxc-project/oxc#9040]
foo.js:

```
    export class Foo {
      #bar;
    
      constructor() {
        this.#bar = 10;
      }
    }
```

This is an invalid warning:

```
x.js:2:3: 'bar' is defined but never used. [Warning]

1 problem
```

Eslint doesn't have this issue

**Comment by overlookmotel** [comment]
I think it's arguable whether that's correct or not. It is pretty pointless to assign to `#bar` property, but never read it. A minifier could correctly reduce that code to:

```js
export class Foo {
  constructor() {
    10;
  }
}
```

**Comment by brupelo** [comment]
Hi guys,

Thanks for taking a look to this one... I've tested further just to make sure what oxlint is doing makes sense, here's my session

```
(venv) D:\sources\github\oxc-project\oxc-bugs>npx oxlint --version
Version: 0.15.10

(venv) D:\sources\github\oxc-project\oxc-bugs>just lint-eslint foo.js
npx eslint foo.js
Warning: React version not specified in eslint-plugin-react settings. See https://github.com/jsx-eslint/eslint-plugin-react#configuration .

D:\sources\github\oxc-project\oxc-bugs\foo.js
  2:3  error  '#bar' is defined but never used  no-unused-private-class-members

✖ 1 problem (1 error, 0 warnings)

error: Recipe `lint-eslint` failed on line 5 with exit code 1

(venv) D:\sources\github\oxc-project\oxc-bugs>just lint-oxlint foo.js
npx oxlint -f unix --fix --fix-suggestions foo.js -c .oxlintrc.json

(venv) D:\sources\github\oxc-project\oxc-bugs>..\oxc\target\release\oxlint --version
Version: dev

(venv) D:\sources\github\oxc-project\oxc-bugs>..\oxc\target\release\oxlint -f unix --fix --fix-suggestions foo.js -c .oxlintrc.json

(venv) D:\sources\github\oxc-project\oxc-bugs>..\oxc\target\release\oxlint -f unix --fix --fix-suggestions foo.js -c .eslintrc.json
foo.js:2:3: 'bar' is defined but never used. [Warning]

1 problem
```

I guess my bug report wasn’t very accurate. I’ll make sure to include the full config files and versions of both Oxlint and ESLint next time I report an issue.  

@overlookmotel Yeah, that’d be great! Not sure if you can minify that easily with existing minifiers though, can you? Actually, this is probably a question for a different forum. Today, I was trying to migrate from Prettier and ESLint to Oxlint, but one thing holding me back was the numerous unused variables that would need to be prefixed with `_`. I couldn’t find a way to do this automatically.  

This was thousands of issues across a large codebase, so the best I could do was add `off` rules in Oxlint to avoid errors and get an `exit_code=0`. For example:  

```js
function foo(bar) {} // ❌
```
Instead of:  
```js
function foo(_bar) {} // ✅
```

and i was trying to use all potential fixes, ie:

```
Fix Problems
        --fix                 Fix as many issues as possible. Only unfixed issues are reported in
                              the output
        --fix-suggestions     Apply auto-fixable suggestions. May change program behavior.
        --fix-dangerously     Apply dangerous fixes and suggestions.
```

**Comment by camc314** [comment]
yeah disabling this rule is probably best for now (`-A no-unused-vars` as a cli arg), that way you don't have to rename very single unused var
